### DISCUSSION TITLE
noob: would switch~ be hard to implement?

### DISCUSSION THREAD

**Discussion by sensorium** [Wasted-Audio/hvcc#191] [Q&A] [answered]
Hello,
please forgive me if this is a ridiculous question, I've only been looking at the hvcc code for a short while and don't know much about it.
Pd's switch~ object would make a big difference to the efficiency of an abstraction I'm converting to an lv2 plugin.  Is it unsupported because something like changing the signal graph (I hope that means something!) on the fly is not practical, or is too complicated?  Or does it just require some time, and a bit of help from others who are familiar with hvcc?
Thanks, hvcc is a great project.

**Comment by dromer** [comment, marked as answer]
Considering the entire architecture of Heavy this is going to be either extremely difficult or simply impossible.

Without some serious DSP experts getting involved with the project, willing to look very deep under the hood, this will likely never happen.

Remember that Heavy only emulates certain PD behavior. We are not using any PD code internally and the entire Heavy C library and runtime works completely different from PD.

**Reply by sensorium** [reply]
OK, thanks for the answer.
